vision/cloud-client/detect/detect.py

```diff
@@ -502,27 +579,34 @@ def detect_document(path):
     with io.open(path, 'rb') as image_file:
         content = image_file.read()
 
-    image = types.Image(content=content)
+    image = vision.types.Image(content=content)
 
     response = client.document_text_detection(image=image)
-    document = response.full_text_annotation
 
-    for page in document.pages:
+    for page in response.full_text_annotation.pages:
         for block in page.blocks:
             block_words = []
             for paragraph in block.paragraphs:
                 block_words.extend(paragraph.words)
+                print(u'Paragraph Confidence: {}\n'.format(
+                    paragraph.confidence))
 
+            block_text = ''
```

Comment:
@dizcology I really, really struggle to easily understand what this sample is doing.

It's not easy on the eyes & doesn't read like prose on the website IMO

Do you really like this sample? Any ideas for ways to highlight parts [TextAnnotation](https://cloud.google.com/vision/docs/reference/rpc/google.cloud.vision.v1#google.cloud.vision.v1.TextAnnotation) that developers want to know about in a more consumable, easy to read sample?

Reply:
For Ruby, I'm rolling with this:

```ruby
  # image_path = "Google Cloud Storage URI, eg. 'gs://my-bucket/image.png'"

  require "google/cloud/vision"

  vision = Google::Cloud::Vision.new

  document = vision.document_text_detection(image_path).full_text_annotation

  puts "Full document text: #{document.text}"

  document.pages.each do |page|
    page.blocks.each  do |block|
      block.paragraphs.each do |paragraph|
        puts "Paragraph confidence: #{paragraph.confidence}"
        paragraph.words.each do |word|
          puts "\tWord confidence: #{word.confidence}"
          puts "\tWord text: #{word.symbols.map(&:text).join}"
          word.symbols.each do |symbol|
            puts "\t\tSymbol text: #{symbol.text}"
            puts "\t\tSymbol confidence: #{symbol.confidence}"
          end
        end
      end
    end
  end
```

I almost don't want to include anything that concatenates/joins text together to print it out (developers can figure that out?) ... but, at least in Ruby it's idiomatic and relatively readable to print out each word:

```ruby
puts "\tWord text: #{word.symbols.map(&:text).join}"
```

The Python snippet includes:
 - 2 Lists used to build up data to print
 - 2 Strings used to build data to print
 - Logic to `extend` Lists
 - Also, 2 types of syntax are used to concatenate strings:

```python
word_text = word_text + symbol.text
```

```python
block_text += ' ' + word_text
```

^--- both should use `+=` or not use `+=` (should be consistent)

*Thoughts?*

Reply:
**Also:** I vote that the sample shows how to print out the WHOLE text without having to concatenate symbols⇒words⇒paragraphs⇒blocks⇒pages

^---- *also, what is a block? ... I hope the docs make that clear, because it's not obvious to me ...*

Reply:
ping @dizcology for thoughts

Also, would love to get this ready for merge ~ this comment is my only remaining question IIRC 😄